clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -1010,23 +977,23 @@ private Fetch<K, V> pollForFetches(Timer timer) {
      *             defined
      * @return true iff the operation completed without timing out
      */
-    private boolean updateFetchPositions(final Timer timer) {
+    private boolean updateFetchPositions() {
         // If any partitions have been truncated due to a leader change, we need to validate the offsets
-        ResetPositionsApplicationEvent event = new ResetPositionsApplicationEvent();
-        eventHandler.add(event);
+        ValidatePositionsApplicationEvent validatePositionsEvent = new ValidatePositionsApplicationEvent();
+        eventHandler.add(validatePositionsEvent);
 
         cachedSubscriptionHasAllFetchPositions = subscriptions.hasAllFetchPositions();
         if (cachedSubscriptionHasAllFetchPositions) return true;
 
```

Comment:
The only thing I see it is missing from the original implementation is 
```
        if (coordinator != null && !coordinator.refreshCommittedOffsetsIfNeeded(timer)) return false;
```

This is a blocking operation: Did we remove this intentionally?

Reply:
it is essentially calling fetch commit offset

Reply:
yes, I left it out intentionally assuming we don't want any group coordination logic for now. Expecting to add it when we integrate the ConsumerCoordinator logic.

Reply:
Created [this](https://confluentinc.atlassian.net/browse/KJC-739) task to track it and add the logic to the coordinatorRequestManager and then use it from there. I can take it up in a following PR after reviewing the details of how to fit this into the CoordinatorReqManager. wdyt?

Reply:
I think you can fetch the last committed position via the `addOffsetFetchRequest` API, then add the epoch update and position seek logic.

Reply:
Agree, I would expect the implementation goes along those lines. But then that CoordinatorManager `addOffsetFetchRequest` functionality needs to be called from the Consumer. Coordinator explicit calls are not integrated yet into the AsyncConsumer, not for any of the functions needing coordination actions (assign, unsubscribe, and this updateFetchPositions) (All simple integration via coordinator events I expect but not included in this first attempt for simple consume).

Reply:
I thought the application thread just needs to fetch the committed offset, and trigger the metadata update and update the subscription state right? once the offset fetch is completed.

Reply:
uhm am I missing something maybe, agree on what should be done, but that needs to be invoked in the consumer only when the consumer is in a group, and that's what I'm referring to. Is that correct? does this only apply if the consumer is in a group?

Anyways, if you think it's better to have it added now I wont merge yet, migrate the `ConsumerCoordinator.refreshCommittedOffsetsIfNeeded` using the existing CommitRequestManager, and then integrate the logic in the AsyncConsumer for calling that if the consumer is in a group.

Reply:
Agreed in offline sync to keep this without making use of committed offsets, as it is not needed for having a simple consume scenario running. Using Kafka-based offset management strategy will be introduced a follow up. Task mentioned above and TODO added here.